### PR TITLE
Add three new arithmetic operators: power, modulo, and integer division

### PR DESCRIPTION
**Pull request by aboyoun**
This PR adds power, modulo, and integer binary operators to the existing arithmetic operators of addition, subtraction, multiplication, and division. Both power and modulo have corresponding functions in the C++ standard library, while integer division follows the definition within R.

### CODE REVIEW

File: include/tatami/isometric/arith_utils.hpp [hunk 1]
```diff
@@ -45,6 +48,25 @@ void delayed_arith_run(Value_& val, Scalar_ scalar) {
...
+    } else if constexpr(op_ == DelayedArithOp::MODULO) {
+        if constexpr(right_) {
+            val = std::fmod(val, scalar);
+        } else {
+            val = std::fmod(scalar, val);
+        }
+    } else if constexpr(op_ == DelayedArithOp::INTEGER_DIVIDE) {
+        // x == (x %% y) + y * (x %/% y)
+        if constexpr(right_) {
+            val = (val - std::fmod(val, scalar)) / scalar;
```
Comment (LTLA) [inline include/tatami/isometric/arith_utils.hpp:66]: Couldn't we just do `std::trunc(val/scalar)`?

File: include/tatami/isometric/binary/arith_helpers.hpp [hunk 2]
```diff
@@ -27,7 +27,9 @@ struct DelayedBinaryArithHelper {
     /**
      * @cond
      */
-    static constexpr bool always_sparse = (op_ != DelayedArithOp::DIVIDE);
+    static constexpr bool always_sparse = (op_ == DelayedArithOp::ADD ||
+                                           op_ == DelayedArithOp::SUBTRACT ||
+                                           op_ == DelayedArithOp::MULTIPLY);
```
Comment (LTLA) [inline include/tatami/isometric/binary/arith_helpers.hpp:32]: Seems like `POWER` would be okay here. The rule of thumb is whether `OP(0, 0) == 0`.

File: include/tatami/isometric/unary/arith_helpers.hpp [hunk 3]
```diff
@@ -63,7 +63,8 @@ Value_ delayed_arith_zero(Scalar_ scalar) {
 template<DelayedArithOp op_, bool right_, typename Value_, typename Scalar_>
 constexpr bool delayed_arith_always_dense() {
     // If we're dividing the scalar by the matrix, values of zero in the matrix will yield non-zero results.
-    if constexpr(op_ == DelayedArithOp::DIVIDE && !right_) {
+    if constexpr((op_ == DelayedArithOp::DIVIDE || op_ == DelayedArithOp::POWER ||
+                  op_ == DelayedArithOp::MODULO || op_ == DelayedArithOp::INTEGER_DIVIDE) && !right_) {
```
Comment (LTLA) [inline include/tatami/isometric/unary/arith_helpers.hpp:67]: Clean this up with:

```cpp
if constexpr(!right_) {
   if constexpr(op == ...) {
       return true;   
   } else if constexpr(op == ...) {
       return true;   
   } etc.
}
```

File: include/tatami/isometric/binary/arith_helpers.hpp (hunk 2 above)
Comment (LTLA) [reply include/tatami/isometric/binary/arith_helpers.hpp:32]: (Same could be said for `delayed_arith_always_sparse`, as long as `POWER` and `right_ = true` without special values.

File: include/tatami/isometric/arith_utils.hpp [hunk 4]
```diff
@@ -45,6 +48,25 @@ void delayed_arith_run(Value_& val, Scalar_ scalar) {
         } else {
             val = scalar / val;
         }
+    } else if constexpr(op_ == DelayedArithOp::POWER) {
+        if constexpr(right_) {
+            val = std::pow(val, scalar);
```
Comment (LTLA) [inline include/tatami/isometric/arith_utils.hpp:53]: Stick a `#include <cmath>` at the top here.

File: include/tatami/isometric/arith_utils.hpp (hunk 1 above)
Comment (aboyoun) [reply include/tatami/isometric/arith_utils.hpp:68]: `std::floor(x / y)` seems to do the trick

Comment (aboyoun) [reply include/tatami/isometric/arith_utils.hpp:68]: Done

File: include/tatami/isometric/arith_utils.hpp [hunk 5]
```diff
@@ -45,6 +50,25 @@ void delayed_arith_run(Value_& val, Scalar_ scalar) {
...
+            val = std::pow(scalar, val);
+        }
+    } else if constexpr(op_ == DelayedArithOp::MODULO) {
+        if constexpr(right_) {
+            val = std::fmod(val, scalar);
+        } else {
+            val = std::fmod(scalar, val);
+        }
+    } else if constexpr(op_ == DelayedArithOp::INTEGER_DIVIDE) {
+        // x == (x %% y) + y * (x %/% y)
```
Comment (LTLA) [inline include/tatami/isometric/arith_utils.hpp:66]: might as well remove this confusing comment.

**Pull request merged**